0.preprocess-sites/8.visualize-plate-layout.py

```diff
@@ -0,0 +1,392 @@
+parser.add_argument(
+    "--config_file",
+    help="configuration yaml file for preprocessing pipeline",
+    default="site_processing_config.yaml",
+)
+args = parser.parse_args()
+config_file = args.config_file
+
+config = process_config_file(config_file)
+
+# Defines the sections of the config file
+core_args = config["core"]
+spots_args = config["process-spots"]
+summ_cell_args = config["summarize-cells"]
+summ_plate_args = config["summarize-plate"]
+
+# Defines the variables set in the config file
+batch = core_args["batch"]
+quality_func = core_args["categorize_cell_quality"]
+ignore_files = core_args["ignore_files"]
+acquisition_side = core_args["acquisition_side"]
+batch_dir = core_args["batch_dir"]
+
+cell_filter = spots_args["cell_filter"]
+
+summ_cells_results_basedir = summ_cell_args["output_resultsdir"]
+summ_cells_figures_basedir = summ_cell_args["output_figuresdir"]
+output_figuresdir = pathlib.Path(summ_cells_figures_basedir, batch)
+cell_category_order = summ_cell_args["cell_category_order"]
+
+correlation_threshold = summ_plate_args["correlation_threshold"]
+painting_image_names = summ_plate_args["painting_image_names"]
+barcoding_cycles = summ_plate_args["barcoding_cycles"]
+barcoding_prefix = summ_plate_args["barcoding_prefix"]
+
+cell_count_file = pathlib.Path(
+    summ_cells_results_basedir, batch, "cells", "cell_count.tsv"
+)
+cell_count_df = pd.read_csv(cell_count_file, sep="\t")
+
+figures_output = pathlib.Path(summ_cells_figures_basedir, batch)
+os.makedirs(figures_output, exist_ok=True)
+results_output = pathlib.Path(summ_cells_results_basedir, batch)
+os.makedirs(results_output, exist_ok=True)
+
+# creates x, y coordinates for plotting per-plate views.
+# assumes image numbering starts in upper left corner and proceeds down
+x_seq = 1
+y_seq = 1
+build_seq = []
+final_order = []
+for i in range(1, ((acquisition_side * acquisition_side) + 1)):
+    if len(final_order) == (acquisition_side * acquisition_side):
+        break
+    else:
+        build_seq.append([x_seq, y_seq])
+        if len(build_seq) == acquisition_side:
+            x_seq = x_seq + 1
+        if y_seq == acquisition_side:
+            y_seq = 0
+            build_seq = build_seq[::-1]
+            final_order.extend(build_seq)
+            build_seq = []
+        y_seq = y_seq + 1
+sites_list = [*range(1, (acquisition_side * acquisition_side) + 1)]
+
+# uses sites_list in case there are fewer analyzed sites than acquired sites
+loc_df = pd.DataFrame(final_order).rename(columns={0: "x_loc", 1: "y_loc"})
+loc_df["Site"] = sites_list
+
+# create total_cell_count
+cell_count_bysite_df = (
+    cell_count_df.groupby("site")["cell_count"]
+    .sum()
+    .reset_index()
+    .rename(columns={"cell_count": "total_cell_count"})
+)
+# add total_cell_count to cell_count_df
+cell_count_df = cell_count_df.merge(cell_count_bysite_df, on="site")
+# add in x, y coordinates for plotting
+cell_count_df = cell_count_df.merge(loc_df, left_on="Site", right_on="Site")
```

Comment:
the capitalization looks off to me given the lower case `site` in line 88 and the uppercase `Site` in line 83. Both upper and lowercase columns store the same info right? We should only use one or the other, especially since we can control that in this script.

Two other points:

- We can also try chaining 94 and 96 
- When the column name is the same, we can use `on="site"` (as you've done in line 94)

Reply:
No, `site` and `Site` are actually different and used differently. `site` is the full string (e.g. 151A1-B2-32) and that is parsed into separate `Plate` `Well` `Site` columns. So `site` is a single unique identifier for each site while `Site` is used for situations where each `Site` is the same regardless of plate and well (e.g. x,y plate coordinates are the same for each `Site` regardless).

Reply:
nice! Is it possible to change `Site` to `Site_Full` instead? I only suggest this change if its entirely internal to this script. Otherwise, we can live with it.

Reply:
It's not entirely internal. We use the same `site` vs `Site` distinction in 7.visualize-cell-summary. But I can change it there too so we have `site_full` that parses into `Plate` `Well` `Site`?

Reply:
👍 if that is the only other place, lets do it!